atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/knowledge/KnownConcludedTransactionsStoreConcurrencyTest.java

```diff
@@ -86,6 +86,31 @@ public void batchesReadsUnderHighConcurrency() throws InterruptedException {
                 .get(eq(TransactionConstants.KNOWN_CONCLUDED_TRANSACTIONS_TABLE), anyMap());
     }
 
+    @Test
+    public void writesPreserveCorrectnessUnderHighConcurrencyAndMinimumTimestampSet() throws InterruptedException {
+        long minimumConcludableTimestamp = 10L;
+        int numThreads = 300;
+        List<Range<Long>> candidateTimestampRanges = LongStream.range(1, numThreads)
+                .mapToObj(index -> Range.closed(2 * index, 2 * index + 1))
+                .filter(range -> !range.isConnected(Range.closed(Long.MIN_VALUE, minimumConcludableTimestamp)))
```

Comment:
hmm i wound not filter here and get a list of all possible ranges. then have a separate filtered list and assert on that.

Reply:
as basically we are asserting that a subset of updates went through, which is fair, but we could let the filter be tester here as well.

Reply:
yup yup makes sense, my initial read of the code was wrong!